marlowe-dashboard-client/src/Page/Dashboard/State.purs

```diff
@@ -384,6 +395,29 @@ handleAction input@{ currentSlot, tzOffset } (ContractAction followerAppId contr
     _ -> toContract followerAppId $ Contract.handleAction contractInput contractAction
 
 ------------------------------------------------------------
+-- Note [polling updateTotalFunds]
+-- We used to have a WebSocket notification whenever the wallet assets changed but that was
+-- removed from the Mock Wallet (see link below) and the WalletBackEnd doesn't provide such
+-- a mechanism anyways.
+-- https://github.com/input-output-hk/plutus-apps/pull/50
+-- Besides that, the WBE has the notion of a wallet being synced, so it makes sense to eventually
+-- switch from WebSocket push to a rest Pull. The reason I didn't do this change in the PR I
+-- introduced this code is that the wallet information is scattered between the Dashboard, Contract
+-- and Contact page.
+-- TODO: SCP-3211 After Halogen Store is merged and once we refactor wallet state into a global store
+--       Refactor this code and do a global polling mechanism.
+updateTotalFunds ::
+  forall m slots msg.
+  MonadAff m =>
+  ManageWallet m =>
+  HalogenM State Action slots msg m Unit
+updateTotalFunds = do
+  walletId <- use (_walletDetails <<< _walletInfo <<< _walletId)
+  response <- getWalletTotalFunds walletId
+  for_ response \(GetTotalFunds { assets }) ->
```

Comment:
We can shorten this to [`Lens.Setter.assign`](https://github.com/purescript-contrib/purescript-profunctor-lenses/blob/v7.0.1/src/Data/Lens/Setter.purs#L96) like above or even just [`.=`](https://github.com/purescript-contrib/purescript-profunctor-lenses/blob/v7.0.1/src/Data/Lens/Setter.purs#L45)

Reply:
Yeah, the only caveat I see is that if we want to modify 2 properties we need to change the "structure" of how we set things. I mean for a single property use assign, for multiple using modify_ + set, why not just use the latter.

Reply:
I'm not sure I get it... you can always:

```purescript
do
  _lens1 := x
  _lens2 := y
```

Reply:
yes, but I'm almost sure that doing 

```purescript
do
  _lens1 := x 
  _lens2 := y
  _lens3 := z
```

causes 3 renders in Halogen, while doing

```purescript
modify_ $ 
   set _lens1 x 
   <<< set _lens2 y 
   <<< set _lens3 z
```

only triggers 1 render.